OsmAnd/src/net/osmand/plus/views/layers/RouteLayer.java

```diff
@@ -150,6 +161,14 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
 
 			final QuadRect latlonRect = cp.getLatLonBounds();
 			final QuadRect correctedQuadRect = getCorrectedQuadRect(latlonRect);
+
+			if (view.hasMapRenderer()) {
+				//OpenGL
+				drawRouteSegment(tileBox);
```

Comment:
Looks like it needs optimisation. OpenGL should be called once per route line calculated / cancelled. Also it should not depend on tileBox.